src/appfl/comm/mpi/mpi_client_communicator.py

```diff
@@ -58,13 +86,57 @@ def get_global_model(self, **kwargs) -> Union[Union[Dict, OrderedDict], Tuple[Un
         else:
             return model, meta_data
         
-    def update_global_model(self, local_model: Union[Dict, OrderedDict, bytes], **kwargs) -> Tuple[Union[Dict, OrderedDict], Dict]:
+    def update_global_model(
+        self, 
+        local_model: Union[Dict, OrderedDict, bytes], 
+        client_id: Optional[Union[str, int]] = None,
+        **kwargs
+    ) -> Tuple[Union[Dict, OrderedDict], Dict]:
         """
-        Send local model to FL server for global update, and return the new global model.
-        :param local_model: the local model to be sent to the server for gloabl aggregation
-        :param kwargs: additional metadata to be sent to the server
-        :return: the updated global model with additional metadata. Specifically, `meta_data["status"]` is either "RUNNING" or "DONE".
+        Send local model(s) to the FL server for global update, and return the new global model.
+        
+        :param local_model: the local model to be sent to the server for global aggregation
+        
+            - `local_model` can be a single model if one MPI process has only one client or one MPI process 
+                has multiple clients but the user wants to send one model at a time
```

Comment:
Just for clarification. When do we have such situation as the following?
> one MPI process has multiple clients but the user wants to send one model at a time

Reply:
Hi @kibaekkim, this [doc](https://github.com/APPFL/APPFL/blob/204-mpi-issue/docs/tutorials/examples_batched_mpi.rst#extra-running-the-batched-mpi-example-for-asynchronous-fl) presents a case.